File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    answer = 0
    arr=[]
    for i in range(1,n+1) : 
        arr.append(i)
        
    print(arr)
    for j in range(0,n+1) :
        summ=0
        k=j        
        while summ <=n and k<n :
            summ+=arr[k]
            k+=1
            if summ == n :
                answer+=1
                break
    return answer

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        
        for(int i =1; i<n+1; i++){
            int sum = i;
            int j = i;
            while(sum < n) {
                j++;
                sum +=j;
            }
            
            if (sum == n) {
                answer ++;
            }
            
        }
        return answer;
    }
}